### PR TITLE
Assignment 1 submission

### PR DESCRIPTION
**Pull request by hahussain5**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Adding code and answering questions. 
## What did you learn from the changes you have made?
Understanding the steps in a KNN classification model and building a model including parameter tuning, using a test and train set, and checking the model's accuracy. 
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
The other approach I thought of when standardizing parameters was to have one dataset that includes both the standardized parameters and the response variable. I thought this would have been more simple when splitting the dataset into a training set and a test set.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
The one part I struggled with was updating the test train split function if the standardized parameters and the response variable were separated. Eventually though looking at the help channel and asking for help I was steered in the right direction. 
## How were these changes tested?
They were tested by running the code. I also plotted the nearest neighbours vs accuracy of the model to see for myself the best parameter in addition to looking at the accuracy grid. 
## Checklist
- [ X] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by KasraVakiloroayaei — CHANGES_REQUESTED** [review]
A1: Incomplete 

Q2part(ii) Also since it is a categorical variable, we do not standardize it 
Q3 - When you use parameter grid, the third argument for the range function, is to skip x values. Thus not training on the entire data frame, thus not finding the most optimal k for KNN. Try removing the third argument to see how the "best n-neighbor" changes. 
Q4 - Please calculate the accuracy once again after making changes finding a new best n-neighbor

**Comment by hahussain5** [comment]
Thank you for the feedback, I did remove the 3rd argument for the parameter grid and now my best n-neighbour value is 23 instead of 21 which was my original result. However, my accuracy did not change once I reran my code again.

**Review by PatelVishakh — CHANGES_REQUESTED** [review]
Assignment 2 missing. No assignment 2 pull request

**Comment by hahussain5** [comment]
Apologies, i had pushed to Git hub but forgot to create the pull request. It should be there now

**Review by KasraVakiloroayaei — APPROVED** [review]
A1: Complete
